shelley/chain-and-ledger/shelley-spec-ledger-test/src/Test/Shelley/Spec/Ledger/Generator/Utxo.hs

```diff
@@ -363,36 +382,36 @@ genNextDelta
       ksIndexedPayScripts
     }
   tx
-  delta@(Delta dfees extraInputs extraWitnesses change _ _) =
-    let baseTxFee = minfee pparams tx
-        encodedLen x = fromIntegral $ BSL.length (serialize x)
+  _count -- the counter of the fix loop
+  delta@(Delta dfees extraInputs extraWitnesses change _ extraScripts) =
+    let !baseTxFee = minfee pparams tx
         -- based on the current contents of delta, how much will the fee
         -- increase when we add the delta to the tx?
         draftSize =
           sum
-            [ 5 :: Integer,  -- safety net in case the coin or a list prefix rolls over into a larger encoding
-              -- Fudge factor, Sometimes we need extra buffer when minting tokens.
-              -- 20 has been empirically determined to make non failing Txs
-              20 :: Integer,
+            [ 1000 :: Integer,  -- safety net in case the coin or a list prefix rolls over into a
+              -- larger encoding, or some other fudge factor occurs. Sometimes we need extra buffer
+              -- when minting tokens. 1000 has been empirically determined to make non-failing Txs
               encodedLen (max dfees (Coin 0)) - 1,
               foldr (\a b -> b + encodedLen a) 0 extraInputs,
               encodedLen change,
               encodedLen extraWitnesses
             ]
-        deltaFee = draftSize <×> Coin (fromIntegral (getField @"_minfeeA" pparams))
-                   <+> Coin (fromIntegral (getField @"_minfeeB" pparams))  -- This is usually very small, so might not have much effect.
+        deltaScriptCost = foldr accum (Coin 0) extraScripts
+            where accum (s1,_) ans = genEraScriptCost @era pparams s1 <+> ans
+        deltaFee = ((draftSize * 1) <×> Coin (fromIntegral (getField @"_minfeeA" pparams))) <+> deltaScriptCost
+                   -- <+> Coin (fromIntegral (getField @"_minfeeB" pparams))  -- This is usually very small, so might not have much effect.
```

Comment:
maybe remove this comment if we think we don't want to account for `b`?

Reply:
will do.

Reply:
done